src/idl_parser.cpp

```diff
@@ -613,6 +605,17 @@ CheckedError Parser::ParseField(StructDef &struct_def) {
                     type.enum_def->underlying_type, &typefield));
   }
 
+  if (type.base_type == BASE_TYPE_VECTOR && type.element == BASE_TYPE_UNION) {
+    // Only cpp supports the union vector feature so far.
+    assert(opts.lang_to_generate == IDLOptions::kCpp);
```

Comment:
has to be an error instead of assert, since it comes from flags supplied by user?